source/patterns/@aws-solutions-constructs/core/lib/opensearch-helper.ts

```diff
@@ -0,0 +1,306 @@
+/**
+ *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import * as opensearch from 'aws-cdk-lib/aws-opensearchservice';
+import { DefaultOpenSearchCfnDomainProps } from './opensearch-defaults';
+import { consolidateProps, addCfnSuppressRules } from './utils';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cdk from 'aws-cdk-lib';
+import * as cloudwatch from 'aws-cdk-lib/aws-cloudwatch';
+import * as cognito from 'aws-cdk-lib/aws-cognito';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from 'constructs';
+
+const MAXIMUM_AZS_IN_OPENSEARCH_DOMAIN = 3;
```

Comment:
Arg - I'm having flashbacks to how bogged down we got in AZs for ElasticSearch...